astropy/samp/tests/test_web_profile.py

```diff
@@ -24,13 +24,14 @@ def setup_module(module):
 
 class TestWebProfile(BaseTestStandardProfile):
 
-    def setup_method(self, method):
+    @pytest.fixture(autouse=True)
+    def setup_method(self, tmp_path):
 
         self.dialog = AlwaysApproveWebProfileDialog()
         t = threading.Thread(target=self.dialog.poll)
         t.start()
 
-        self.tmpdir = tempfile.mkdtemp()
+        self.tmpdir = str(tmp_path)
         lockfile = os.path.join(self.tmpdir, '.samp')
```

Comment:
This is the only use of `os` in this file, so we can avoid importing it.
```suggestion
        lockfile = tmp_path / ".samp"
```

Reply:
I am not sure if the lockfile can take a Path object.

Reply:
Good point. I did have that earlier and then I got derailed and I reverted and forgot about it.